### ISSUE TITLE
Documentation: Empty hitsPerPageSelector default value

### ISSUE THREAD

**Issue by pixelastic** [algolia/instantsearch.js#504]
On first load, the hitsPerPageSelector widget example has no value selected. It should have one.

![image](https://cloud.githubusercontent.com/assets/283419/11029637/8ce4a6a2-86c8-11e5-96c5-a7b5f0501cba.png)

**Comment by pixelastic** [comment]
As we now have independent widgets in the documentation, it means that no `hits` widget were loaded before the `hitsPerPageSelector` that could set the `hitsPerPage` value of the helper state.

The `hitsPerPageSelector` currently silently ignore the error, and thus does not auto-select any of the passed values. If a `hitsPerPage` exists, it select the corresponding value in the dropdown or throw an error if none matches.

I suggest that we change this behavior and that if no `hitsPerPage` is previously set, we set it to the first value of the dropdown. We'll have to make sure the order in which `hits` and `hitsPerPageSelector` are started is irrelevant.

In a nutshell, here what I suggest, in pseudo-code:

```javascript
// Case 1
hitsWidget(hitsPerPage: 6);
hitsPerPageWidget(); // Select the value 6 if in the list, throw an error if no such value

// Case 2
hitsPerPageWidget(); // Select the first value of the list and set it as hitsPerPage

// Case 3
hitsPerPageWidget(); // Select first value of the list and set it as hitsPerPage
hitsWidget(hitsPerPage: 6); // Changes the hitsPerPage value. Updates the dropdown selected value if in the list, throw an error otherwise
```

WDYT?







- `hitsPerPage` set in the `hits` widget

**Comment by vvo** [comment]
WFM, wonder if that a hard change

**Comment by redox** [comment]
:+1:

**Comment by bobylito** [comment]
There is always a hitsperpage either explicitly or implicitly so we can't just choose a value in the list provided by the user... So I would that this solution will lead to inconsistency. The best we can do is update the value from the results of the search.

**Comment by redox** [comment]
> There is always a hitsperpage either explicitly or implicitly so we can't just choose a value in the list provided by the user... So I would that this solution will lead to inconsistency. The best we can do is update the value from the results of the search.

Even if you have a value in the index settings (so "implicitly" defined), what you want is to drive the number of hits per page with the values from the select because that's how you've designed your UI: I think @pixelastic's proposition is OK.

**Comment by Jerskouille** [comment]
:+1: @pixelastic and @redox

**Comment by bobylito** [comment]
After discussing it IRL, we think that it's an edge that would only happen if the user doesn't use the hits widget nor define the `hitsPerPage` attribute using the initial search parameters in the instantsearch constructor. So it should be fixed in the documentation and not in the widget.
